File: 2025/c/src/1_1.c
#include "common_function.h"
#include <stdio.h>
#include <stdlib.h>

int main() {
    char *path = "../input/1.txt";
    char *file_content = read_file_to_string(path);
    char **split_result = split_char(file_content, '\n');
    int counter = 50;
    int result = 0;
    for (int i = 0; split_result[i] != NULL; i++) {
        if (split_result[i][0] == '\0') {
            continue;
        }
        char direction = split_result[i][0];
        long force = strtol(split_result[i] + 1, NULL, 10);
        force %= 100;
        if (direction == 'L') {
            counter -= (int)force;
        }
        else if (direction == 'R') {
            counter += (int)force;
        }
        else {
            continue;
        }
        counter %= 100;
        if (counter < 0){
            counter += 100;
        }
        if (counter == 0){
            result++;
        }
    }
    printf("%d\n", result);
    return 0;
}

File: 2025/ts/src/1_1.ts
import * as fs from 'fs';

const input = fs.readFileSync("../input/1.txt", "utf8").split("\r\n");
let counter = 50;
let result = 0;
for(let i = 0; i < input.length; i++) {
    let direction = input[i][0];
    let force = parseInt(input[i].substring(1));
    force %= 100;
    if (direction == 'L') {
        counter -= force;
    }
    else if (direction == 'R') {
        counter += force;
    }
    else {
        continue;
    }
    counter %= 100;
    if (counter < 0){
        counter += 100;
    }
    if (counter == 0){
        result++;
    }
    console.log(counter);
}
console.log(result);